src/track/track.cpp

```diff
@@ -43,17 +42,10 @@ bool compareAndSet(T* pField, const T& value) {
     }
 }
 
-inline mixxx::Bpm getActualBpm(
-        mixxx::Bpm bpm,
-        mixxx::BeatsPointer pBeats = mixxx::BeatsPointer()) {
-    // Only use the imported BPM if the beat grid is not valid!
-    // Reason: The BPM value in the metadata might be normalized
-    // or rounded, e.g. ID3v2 only supports integer values.
-    if (pBeats) {
-        return mixxx::Bpm(pBeats->getBpm());
-    } else {
-        return bpm;
-    }
+inline mixxx::Bpm getBeatsPointerBpm(
```

Comment:
I was almost tempted to dispose the whole function. It is only used at one place and inlineing it there makes the code more easy to understand, without the extra comment IMHO. But I don't mind if you keep it.

Reply:
For the internal use, a member function might be more straight forward, because the default value is always 0.0,

Reply:
Please note that this function is also used when importing new metadata. In this case the imported BPM must be passed as the default.

Reply:
Inlined for the special case of importing metadata.

Reply:
It does not need to be a member function, because it operates only on the BeatsPointer, not the Track itself. A free function is more appropriate. If it is needed elsewhere it could become a static member of `Beats`.